failing to detect invalid assignment when using ? wildcard

### Command

```
javac -J--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED \
  -J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED \
-processor org.checkerframework.checker.nullness.NullnessChecker \
-cp checker-framework/checker/dist/checker.jar -Aignorejdkastub Test.java
```

### File

```java
import java.util.*;
import org.checkerframework.checker.nullness.qual.*;

class A<T> {
    T f;
    A(T f) {this.f = f;}
    T getF() { return this.f; }
}

public class Test {

  static public void main(String[] args) {
      Map<@Nullable String, String> map = new HashMap<@Nullable String, String>();
      map.put(null, "");
      A<Map<@Nullable String, String>> x = new A<Map<@Nullable String, String>>(map);
      Map<String, ?> y = x.getF();
      y.keySet().iterator().next().toString();
  }
}
```

### Actual behavior

The code passes the check, but there's a NPE

```
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "String.toString()" because the return value of "java.util.Iterator.next()" is null
        at Test.main(Test.java:17)
```

### Expected behavior

The code should have been rejected

```
Test.java:16: error: [assignment] incompatible types in assignment.
      Map<String, ?> y = x.getF(); // invalid assignment
                                    ^
```

--- a/checker/tests/nullness/Issue6890.java
+++ b/checker/tests/nullness/Issue6890.java
@@ -1,0 +1,24 @@
+package open.falseneg;
+
+import java.util.Map;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class Issue6890 {
+  static class SomeClass<T> {
+    T f;
+
+    SomeClass(T f) {
+      this.f = f;
+    }
+
+    T getF() {
+      return this.f;
+    }
+  }
+
+  public static void method(Map<@Nullable String, String> map) {
+    SomeClass<Map<@Nullable String, String>> x = new SomeClass<Map<@Nullable String, String>>(map);
+    // :: error: [assignment]
+    Map<String, ?> y = x.getF(); // invalid assignment
+  }
+}

--- a/framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java
+++ b/framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java
@@ -462,7 +462,8 @@ public class DefaultTypeHierarchy extends AbstractAtmComboVisitor<Boolean, Void>
       // calls isSubtype(inside.getUpperBound(), outsideUpper). This is slightly different
       // from the algorithm in the JLS.  Only one of the Java type bounds can be specified,
       // but there can be annotations on both the upper and lower bound of a wildcard.
-      return isSubtype(outsideLower, inside) && isSubtype(inside, outsideUpper);
+      return isSubtype(outsideLower, inside, currentTop)
+          && isSubtype(inside, outsideUpper, currentTop);
     } catch (Throwable ex) {
       // Work around:
       // https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8265255

